platform/posix/clock_posix.c

```diff
@@ -37,24 +37,28 @@
 /*
  * Time conversion constants.
  */
-#define NANOSECONDS_PER_MILLISECOND    ( 1000000 )    /**< @brief Nanoseconds per millisecond. */
-#define MILLISECONDS_PER_SECOND        ( 1000 )       /**< @brief Milliseconds per second. */
+#define NANOSECONDS_PER_MILLISECOND    ( 1000000L )    /**< @brief Nanoseconds per millisecond. */
+#define MILLISECONDS_PER_SECOND        ( 1000L )       /**< @brief Milliseconds per second. */
 
 /*-----------------------------------------------------------*/
 
 uint32_t Clock_GetTimeMs( void )
 {
-    uint32_t timeMs;
+    int64_t timeMs;
     struct timespec timeSpec;
 
     /* Get the MONOTONIC time. */
-    clock_gettime( CLOCK_MONOTONIC, &timeSpec );
+    ( void ) clock_gettime( CLOCK_MONOTONIC, &timeSpec );
 
     /* Calculate the milliseconds from timespec. */
-    timeMs = ( uint32_t ) ( timeSpec.tv_sec * 1000 )
-             + ( uint32_t ) ( timeSpec.tv_nsec / ( 1000 * 1000 ) );
+    timeMs = ( timeSpec.tv_sec * MILLISECONDS_PER_SECOND )
```

Comment:
This is a 32_bit multiplication. Seems like int64_t for timeMs is not required, this way will just use twice the memory with the same precision?